centreon/www/install/php/Update-25.01.0.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * Copyright 2005 - 2024 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+require_once __DIR__ . '/../../../bootstrap.php';
+require_once __DIR__ . '/../../class/centreonLog.class.php';
+
+$centreonLog = CentreonLog::create();
+
+// error specific content
+$versionOfTheUpgrade = 'UPGRADE - 25.01.0: ';
+$errorMessage = '';
+
+$addColumnToResourcesTable = function (CentreonDB $pearDB) use (&$errorMessage): void {
+    $errorMessage = 'Unable to add column flapping to table resources';
+    if (! $pearDB->isColumnExist('resources', 'flapping')) {
+        $pearDB->executeQuery(
+            <<<'SQL'
+                ALTER TABLE `resources`
+                ADD COLUMN `flapping` TINYINT(1) NOT NULL DEFAULT 0
+            SQL
+        );
+    }
+    if (! $pearDB->isColumnExist('resources', 'percent_state_change')) {
+        $pearDB->executeQuery(
+            <<<'SQL'
+                ALTER TABLE `resources`
+                ADD COLUMN `percent_state_change` FLOAT DEFAULT NULL
+            SQL
+        );
+    }
+};
+
+try {
+    $addColumnToResourcesTable($pearDB);
+} catch (Exception $e) {
+    if ($pearDB->inTransaction()) {
+        $pearDB->rollBack();
+    }
+
+    $centreonLog->log(
+        CentreonLog::TYPE_UPGRADE,
+        CentreonLog::LEVEL_ERROR,
+        $versionOfTheUpgrade . $errorMessage
+        . ' - Code : ' . (int) $e->getCode()
+        . ' - Error : ' . $e->getMessage()
+        . ' - Trace : ' . $e->getTraceAsString()
+    );
```

Comment:
use new CentreonLog Methods like error() please and pass the exception in option.